tests/views/test_accounts.py

```diff
@@ -171,219 +172,141 @@ def test_name_uaa_error(self, mock_request):
-            mock_request.post(url_uaa_token, json={"access_token": self.access_token}, status_code=201)
-            mock_request.get(url_uaa_user_by_id, json=uaa_user_by_id_json, status_code=200)
-            mock_request.put(url_uaa_user_by_id, status_code=200)
-            response = self.client.post(
-                "/account/change-username",
-                follow_redirects=True,
-                data={"username": ""},
-            )
-            self.assertIn(b"Username is required", response.data)
+        response = self.client.get("/account/change-email", follow_redirects=True)
 
-    @requests_mock.mock()
-    def test_username_is_max_name_length(self, mock_request):
-        with patch("response_operations_ui.views.accounts.NotifyController") as mock_notify:
-            with self.client.session_transaction() as session:
-                session["user_id"] = user_id
-            mock_notify()._send_message.return_value = mock.Mock()
-            mock_request.post(url_uaa_token, json={"access_token": self.access_token}, status_code=201)
-            mock_request.get(url_uaa_user_by_id, json=uaa_user_by_id_json, status_code=200)
-            mock_request.put(url_uaa_user_by_id, status_code=200)
-            response = self.client.post(
-                "/account/change-username",
-                follow_redirects=True,
-                data={"username": max_256_characters},
-            )
-            self.assertIn(b"Username must be less than 255 characters", response.data)
+        self.assertIn(b"New email address", response.data)
+        self.assertIn(b"Re-type new email address", response.data)
+        self.assertEqual(response.status_code, 200)
 
     @requests_mock.mock()
-    def test_username_contains_wrong_chars(self, mock_request):
+    def test_change_email_request(self, mock_request):
         with patch("response_operations_ui.views.accounts.NotifyController") as mock_notify:
             with self.client.session_transaction() as session:
                 session["user_id"] = user_id
             mock_notify()._send_message.return_value = mock.Mock()
             mock_request.post(url_uaa_token, json={"access_token": self.access_token}, status_code=201)
             mock_request.get(url_uaa_user_by_id, json=uaa_user_by_id_json, status_code=200)
-            mock_request.put(url_uaa_user_by_id, status_code=200)
-            response = self.client.post(
-                "/account/change-username",
-                follow_redirects=True,
-                data={"username": "uaa_user!"},
-            )
-            self.assertIn(
-                b"Username can only contain lowercase letters, numbers, and special characters (`.`, `@`, and `_`)",
-                response.data,
-            )
 
-    @requests_mock.mock()
-    def test_username_max_length_and_wrong_character_errors(self, mock_request):
-        with patch("response_operations_ui.views.accounts.NotifyController") as mock_notify:
```

Comment:
There is loads of unnecessary mocking in these tests, which just adds noise and makes them harder to read. This would fail on form validation and never get near notify, so there is no need to mock it. I think I removed 40 lines of code in 4 tests :upside_down_face: